### PR TITLE
Make the notification slider work

### PR DESCRIPTION
**Pull request by dbkr**
* Changes the rule kind of the display name rule to 'override' as per https://github.com/matrix-org/matrix-doc/pull/373
 * Use new functionality in react-sdk to get & set the 2 other rule states (and make the current 2 work as expected)

Requires https://github.com/matrix-org/matrix-react-sdk/pull/420

### CODE REVIEW

File: src/components/views/context_menus/NotificationStateContextMenu.js [hunk 1]
```diff
@@ -31,95 +31,78 @@ module.exports = React.createClass({
...
     },
 
-    _save: function( areNotifsMuted ) {
-        var self = this;
+    _save: function(newState) {
+        const oldState = this.state.roomNotifState;
         const roomId = this.props.room.roomId;
         var cli = MatrixClientPeg.get();
 
         if (!cli.isGuest()) {
```
Comment (richvdh) [inline src/components/views/context_menus/NotificationStateContextMenu.js]: invert this condition and return early?

File: src/components/views/context_menus/NotificationStateContextMenu.js [hunk 2]
```diff
@@ -31,95 +31,78 @@ module.exports = React.createClass({
...
             // Wrapping this in a q promise, as setRoomMutePushRule can return
```
Comment (richvdh) [inline src/components/views/context_menus/NotificationStateContextMenu.js]: comment appears to be a lie now

File: src/components/views/context_menus/NotificationStateContextMenu.js [hunk 3]
```diff
@@ -31,95 +31,78 @@ module.exports = React.createClass({
...
+                    if (this.props.onFinished) {
+                        this.props.onFinished();
                     };
                 });
-            }).fail(function(error) {
+            }, (error) => {
                 // TODO: some form of error notification to the user
                 // to inform them that their state change failed.
+                // For now we at least set the state back
+                this.setState({
```
Comment (richvdh) [inline src/components/views/context_menus/NotificationStateContextMenu.js]: could probably do with an unmounted guard

File: src/components/views/context_menus/NotificationStateContextMenu.js [hunk 4]
```diff
@@ -31,95 +31,78 @@ module.exports = React.createClass({
...
-                        action: 'notification_change',
-                        roomId: self.props.room.roomId,
-                        areNotifsMuted: areNotifsMuted,
-                    });
-
+                return q.delay(500).then(() => {
                     // Close the context menu
-                    if (self.props.onFinished) {
-                        self.props.onFinished();
+                    if (this.props.onFinished) {
```
Comment (richvdh) [inline src/components/views/context_menus/NotificationStateContextMenu.js]: could probably do with an unmounted guard

**Comment by dbkr** [comment]
(We should do the thing to make it use the react-sdk branch of the same name if it exists...)

**Comment by richvdh** [comment]
> (We should do the thing to make it use the react-sdk branch of the same name if it exists...)

yeah but omgthefaff

**Comment by richvdh** [comment]
LGTM

**Pull request merged**